src/modules/flutter.rs

```diff
@@ -0,0 +1,256 @@
+use super::{Context, Module, ModuleConfig};
+
+use crate::configs::flutter::FlutterConfig;
+use crate::formatter::StringFormatter;
+use crate::formatter::VersionFormatter;
+use crate::utils::get_command_string_output;
+use yaml_rust::Yaml;
+use yaml_rust::YamlLoader;
+
+/// Creates a module with the current Flutter version
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("flutter");
+    let config = FlutterConfig::try_load(module.config);
+
+    let is_dart_project = context
+        .try_begin_scan()?
+        .set_files(&config.detect_files)
+        .set_extensions(&config.detect_extensions)
+        .set_folders(&config.detect_folders)
+        .is_match();
+
+    if !is_dart_project {
+        return None;
+    }
+
+    // try to read and parse pubspec.yaml and check for flutter dependency
+    if config.check_pubspec {
+        has_flutter_dependency(context)?;
+    }
+
+    let parsed = StringFormatter::new(config.format).and_then(|formatter| {
+        let flutter_output = get_flutter_version(context);
+        formatter
+            .map_meta(|var, _| match var {
+                "symbol" => Some(config.symbol),
+                _ => None,
+            })
+            .map_style(|variable| match variable {
+                "style" => Some(Ok(config.style)),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "version" => VersionFormatter::format_module_version(
+                    module.get_name(),
+                    &flutter_output.clone()?.0,
+                    config.version_format,
+                )
+                .map(Ok),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "channel" => flutter_output.clone().map(|(_, c)| Ok(c)),
+                _ => None,
+            })
```

Comment:
Please only use a single `.map*` function of each type, I also removed the `clone`-calls:

```suggestion
            .map(|variable| match variable {
                "version" => VersionFormatter::format_module_version(
                    module.get_name(),
                    &flutter_output.as_ref()?.0,
                    config.version_format,
                )
                .map(Cow::Owned)
                .map(Ok),
                "channel" => flutter_output
                    .as_ref()
                    .map(|(_, c)| Ok(Cow::Borrowed(c.as_str()))),
                _ => None,
            })
```